[BUG] rdiffbackup module fail to load with captured stderr and stdout

## Bug summary

When loading `rdiffbackup` module on Windows with a fake strerr and stdout (mainly to capture logs during testing with pytest) the module refused to load.

## Version, Python, Operating System

On Windows, all python version.

## rdiff-backup call

```
import rdiffbackup
```

## What happened and what did you expect?

```
    import rdiffbackup.run
.tox\py3-win\lib\site-packages\rdiffbackup\run.py:30: in <module>
    msvcrt.setmode(sys.stdin.fileno(), os.O_BINARY)
.tox\py3-win\lib\site-packages\_pytest\capture.py:220: in fileno
    raise UnsupportedOperation("redirected stdin is pseudofile, has no fileno()")
E   io.UnsupportedOperation: redirected stdin is pseudofile, has no fileno()
```

## More information

I'm proposing to move the call of `msvcrt.setmode` into `main` function to let the module load regardless.

--- a/src/rdiffbackup/run.py
+++ b/src/rdiffbackup/run.py
@@ -23,17 +23,15 @@ import sys
 from rdiff_backup import Globals, log
 from rdiffbackup import arguments, actions_mgr, actions
 
-try:
+if os.name == "nt":
     import msvcrt
-
-    # make sure line endings are kept under Windows like under Linux
-    msvcrt.setmode(sys.stdin.fileno(), os.O_BINARY)
-    msvcrt.setmode(sys.stdout.fileno(), os.O_BINARY)
-except ImportError:
-    pass
 
 
 def main():
+    if os.name == "nt":
+        # make sure line endings are kept under Windows like under Linux
+        msvcrt.setmode(sys.stdin.fileno(), os.O_BINARY)
+        msvcrt.setmode(sys.stdout.fileno(), os.O_BINARY)
     sys.exit(main_run(sys.argv[1:]))
 
 

